extensions/pom.xml

```
@@ -21,52 +21,40 @@
         <module>transports-http-hc5</module>
     </modules>
 
-    <build>
-        <plugins>
-            <plugin>
-                <groupId>org.codehaus.gmaven</groupId>
-                <artifactId>groovy-maven-plugin</artifactId>
-                <executions>
-                    <execution>
-                        <id>set-enforce-non-deployment-property</id>
-                        <inherited>false</inherited>
-                        <goals>
-                            <goal>execute</goal>
-                        </goals>
-                        <phase>process-resources</phase>
-                        <configuration>
-                            <source>
-                                if (project.artifactId.endsWith('-deployment')) {
-                                    properties['enforce-non-deployment.skip'] = true;
-                                }
-                            </source>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-enforcer-plugin</artifactId>
-                <executions>
-                    <execution>
-                        <id>enforce-non-deployment</id>
-                        <configuration>
-                            <skip>${enforce-non-deployment.skip}</skip>
-                            <rules>
-                                <bannedDependencies>
-                                    <excludes>
-                                        <!-- <exclude>org.ow2.asm:asm</exclude> -->
-                                    </excludes>
-                                </bannedDependencies>
-                            </rules>
-                        </configuration>
-                        <goals>
-                            <goal>enforce</goal>
-                        </goals>
-                    </execution>
-                </executions>
-            </plugin>
-        </plugins>
-    </build>
-
+    <profiles>
+        <profile>
+            <id>non-deployment-bans</id>
+            <activation>
+                <file>
+                    <exists>${project.basedir}/src/main/resources/META-INF/quarkus-extension.yaml</exists>
```

```suggestion
                    <exists>${basedir}/src/main/resources/META-INF/quarkus-extension.yaml</exists>
```

That placeholder rang a bell and indeed there's a warning:
```
Warning:  
Warning:  Some problems were encountered while building the effective model for io.quarkiverse.cxf:quarkus-cxf-extensions:pom:1.5.0-SNAPSHOT
Warning:  'profiles.profile[non-deployment-bans].activation.file.exists' Failed to interpolate file location ${project.basedir}/src/main/resources/META-INF/quarkus-extension.yaml for profile non-deployment-bans: ${project.basedir} expression not supported during profile activation, use ${basedir} instead
```

Ran into this myself multiple times.

---

Hm... I knew that profile activation is quite special, but I have not seen this before. Thanks for the hint! I used to use `${basedir}` util I saw that it is going to get removed in Maven 4: https://github.com/apache/maven/pull/696

But OK, let's go back to `${basedir}` for now.

---

Yeah, it's a nasty and confusing pitfall.